### PR TITLE
Add build instructions for RedHat / Debian based OSes

### PR DESCRIPTION
**Pull request by LeoDog896**
This adds instructions to the build for what dependencies to install for RedHat & Debian based systems with Qt 5 / Qt 6 options

### CODE REVIEW

File: src/wiki/development/build-instructions.md [hunk 1]
```diff
@@ -33,17 +33,45 @@ Getting the project to build and run on Linux is easy if you use any modern and
...
+- zlib
+- Java JDK
+- GL headers
+- scdoc if you want to generate manpages
 
 You can use IDEs, like KDevelop, QtCreator or CLion to open the CMake project, if you want to work on the code.
 
+## Distributions
+
+The following distribution instructions are with fresh instances of itself with podman. They are opinionated and use GCC for their C compiler.
```
Comment (DioEgizio) [inline src/wiki/development/build-instructions.md:49]:
```suggestion
Commands to install the dependencies with the following package managers.
```

File: src/wiki/development/build-instructions.md [hunk 2]
```diff
@@ -33,17 +33,45 @@ Getting the project to build and run on Linux is easy if you use any modern and
...
+
+### Fedora, CentOS Stream/RHEL
```
Comment (DioEgizio) [inline src/wiki/development/build-instructions.md:51]:
```suggestion
### DNF (Fedora, CentOS Stream/RHEL)
```

File: src/wiki/development/build-instructions.md [hunk 3]
```diff
@@ -33,17 +33,45 @@ Getting the project to build and run on Linux is easy if you use any modern and
...
+
+```
+sudo dnf install git gcc-c++ cmake java-latest-openjdk-devel zlib-devel extra-cmake-modules qtchooser
+
+# Qt 5
+sudo dnf install qt5-qtbase-devel
```
Comment (DioEgizio) [inline src/wiki/development/build-instructions.md:57]: probably not enough to build polymc

File: src/wiki/development/build-instructions.md [hunk 4]
```diff
@@ -33,17 +33,45 @@ Getting the project to build and run on Linux is easy if you use any modern and
...
+
+# Qt 6
+sudo dnf install qt6-qtbase-devel
```
Comment (DioEgizio) [inline src/wiki/development/build-instructions.md:60]: definetly needs more dependencies for building polymc

File: src/wiki/development/build-instructions.md [hunk 5]
```diff
@@ -33,17 +33,45 @@ Getting the project to build and run on Linux is easy if you use any modern and
...
+### Debian, Ubuntu, Mint
+```
+sudo apt update
+sudo apt install git cmake build-essential openjdk-17-jdk zlib1g-dev extra-cmake-modules libgl1-mesa-dev scdoc
+
+# Qt 5
+sudo apt install qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools libqt5core5a libqt5network5 libqt5gui5
+
+# Qt 6
+sudo apt install
```
Comment (DioEgizio) [inline src/wiki/development/build-instructions.md:72]:
```suggestion
sudo apt install qt6-base-dev qtchooser qt6-base-dev-tools libqt6core6 libqt6core5compat6 libqt6network6
```

File: src/wiki/development/build-instructions.md [hunk 6]
```diff
@@ -33,17 +33,45 @@ Getting the project to build and run on Linux is easy if you use any modern and
...
+sudo dnf install git gcc-c++ cmake java-latest-openjdk-devel zlib-devel extra-cmake-modules qtchooser
+
+# Qt 5
+sudo dnf install qt5-qtbase-devel
+
+# Qt 6
+sudo dnf install qt6-qtbase-devel
+```
+
+### Debian, Ubuntu, Mint
```
Comment (DioEgizio) [inline src/wiki/development/build-instructions.md:63]:
```suggestion
### APT (Debian, Ubuntu, Mint and derivates)
```

File: src/wiki/development/build-instructions.md (hunk 4 above)
Comment (LeoDog896) [reply src/wiki/development/build-instructions.md:60]: That's odd -- it builds perfectly fine on podman. I'll try a VM when I get home to see of the build works there.

Comment (DioEgizio) [reply src/wiki/development/build-instructions.md:60]: well polymc needs core5compat on qt6, soooo

Comment (LeoDog896) [reply src/wiki/development/build-instructions.md:60]: huh, probably ran it wrong then -- I'll be able to check once I get back.

**Comment by ZekeSmith** [comment]
seems good I will need to work out the merge conflicts and I am happy to merge this once the current issue is resolved.

**Pull request closed without merging**